WireAPI/Tests/WireAPITests/APIs/AccountsAPI/AccountsAPITests.swift

```diff
@@ -0,0 +1,124 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+
+@testable import WireAPI
+@testable import WireAPISupport
+
+final class AccountsAPITests: XCTestCase {
+
+    private var apiSnapshotHelper: APIServiceSnapshotHelper<any AccountsAPI>!
+
+    // MARK: - Setup
+
+    override func setUp() {
+        apiSnapshotHelper = APIServiceSnapshotHelper<any AccountsAPI> { apiService, apiVersion in
+            AccountsAPIBuilder(apiService: apiService)
+                .makeAPI(for: apiVersion)
+
+        }
+    }
+
+    override func tearDown() {
+        apiSnapshotHelper = nil
+    }
+
+    // MARK: - Request generation
+
+    func testUpgradeToTeam_V0_To_V6() async throws {
+        // Given
+        let apiService = MockAPIServiceProtocol()
+        let builder = AccountsAPIBuilder(apiService: apiService)
+
+        for apiVersion in [APIVersion.v0, .v1, .v2, .v3, .v4, .v5, .v6] {
+            let sut = builder.makeAPI(for: apiVersion)
+
+            // Then
+            await XCTAssertThrowsErrorAsync(AccountsAPIError.unsupportedEndpointForAPIVersion) {
+                try await sut.upgradeToTeam(teamName: Scaffolding.teamName)
+            }
+        }
+    }
+
+    func testUpgradeToTeam_Request_Generation_V7_Onwards() async throws {
+        // Given
+        let apiVersions = APIVersion.v7.andNextVersions
+
+        // Then
+        try await apiSnapshotHelper.verifyRequest(for: apiVersions) { sut in
```

Comment:
we need: `.outputFormatting = .sortedKeys` in the decoder I guess. We have a JSONEncoder+Default.swift it might be good to do the same here for JSONDecoder if needed

Reply:
There's no `.outputFormatting` for `JSONDecoder`